Windows/MainWindow.UIControls.cs

```diff
@@ -268,13 +269,33 @@ public partial class MainWindow : Window {
         }
 
         private void AppMainWindow_PreviewMouseWheel(object sender, MouseWheelEventArgs e) {
-            if (ctrlKeyDown) {
-                var delta = (int) Helper.DoubleRangeTruncate(e.Delta, -1, 1);
+            if (!ctrlKeyDown) {
+                return;
+            }
+
+            var delta = (int) Helper.DoubleRangeTruncate(e.Delta, -1, 1);
+
+            double pos = gridController.mouseBeat;
+            BPMChange lastChange = mapEditor.GetLastBeatChange(pos);
+
+            if (lastChange.globalBeat == 0.0) {
                 int currentBeatDivision;
                 int.TryParse(txtGridDivision.Text, out currentBeatDivision);
                 txtGridDivision.Text = ((int)Helper.DoubleRangeTruncate(currentBeatDivision + delta, 1, Editor.GridDivisionMax)).ToString();
-                e.Handled = true; // Mark tunneling event as handled to prevent scrolling on the grid while changing the division.
+            } else {    
+                // Update the BPM value in the original list
+                foreach ( BPMChange bpmChange in mapEditor.currentMapDifficulty.bpmChanges.OrderByDescending(bpmChange => bpmChange.globalBeat)) {
```

Comment:
This is more a styling thing and not necessary to fix - I don't like mixing LINQ and standard loops together, as it makes for a harder read. Something like this is easier to understand imho:
```
var currentBpmChange = mapEditor.currentMapDifficulty.bpmChanges
    .Where(bpmChange => bpmChange.globalBeat < pos)
    .OrderByDescending(bpmChange => bpmChange.globalBeat)
    .FirstOrDefault(null);
if (currentBpmChange != null) {
    currentBpmChange.gridDivision = (int) Helper.DoubleRangeTruncate(currentBpmChange.gridDivision + delta, 1, Editor.GridDivisionMax);
    gridController.DrawGrid(false);
}
```

Reply:
I'm still not warm to csharp. I thought this would return a copy of that value, so I decided to wrote it that way.
But thanks for your suggestion, this is definitely better.